### ISSUE TITLE
Bolding of Style labels seems to be inconsistent/flakey

### ISSUE THREAD

**Issue by davidc86** [sillsdev/ptx2pdf#399]
Elsewhere, MH wrote: "There is nothing that says that a parameter's label is bolded due to a difference to some 'parent'. **The tooltip is quite clear: a parameter's label is bolded if the value is different from what is in the stylesheet.** More specifically as specified in the chain of usfm_sb.sty, custom.sty ptxprint-mods.sty. I.e. that what you see is uniquely edited here in ptxprint gui."

"Also, because there is **_no notion of parent styles, there is no sub-nodes either_**. They are just a way of providing a visual hierarchy for finding markers. **So, no setting something in one marker/node does not set anything in any other nodes/markers.** Having said that, we are toying with the idea of doing that for q/q1 and s/s1, but only because they are near synonyms and we would have to think about what happens if someone wants them different."

(emphasis mine)

That is very, very interesting. I had assumed a parent/sub-node idea. I had assumed that \s2 was somehow related to \s1, and so on. I had assumed that setting the "main" \s parameters would propagate through the "children" to save me time. GOOD to KNOW this is NOT the case. (I bet I'm not the only one who will need this spelled out to them.)

As to **Bolding** of Style Labels. My current project I have NO external stylesheets checked. So, I assume that means only usfm.sty is read from the Paratext folder. So then any bolding of style labels I see must be do to differences between my current PTXprint settings and ones that come from the UBS/SIL factory, right? 

**Here is the problem:** I can click on a bold style label, and it resets the value of the setting (to match some hidden thing), and the style label is UN-bolded. But then when I move to another style and come right back, the label is Bold again, and clicking on it does NOT change the value (because it is already correct), but the label goes Un-bold again. This is totally repeatable. 

This is what I meant about the Bold labels not behaving as expected. 
Is it possible it is a rounding error that is causing it to think those values are not exactly like the usfm.sty file's? 
Anyway, this magic re-bolding of labels will confuse people.

The explanation of no-parents and no-children, is good to know: I.e. _Any tweaking has to be done for everything in a group, if we want a consistent look_. E.g. if we want all intro material to be a pt size (or so) smaller than the main text, then EVERY SINGLE INTRO field marker needs to be consistently tweaked to correct the factored value (both Font and line spacing). And all footnotes markers also need to be tweaked as well, if they are to be a smaller size too. (And if we change your mind, we will need to do it all over again, so we need to make sure we have the factor values correct before we start doing all of them.)

Not a complain, just a sober truth about how setting up the style sheet is not as trivial as one might think.

**Comment by mhosken** [comment]
I think I've solved the spurious bolding bug. Of course there may be other causes I may have missed. If so, let me know the specific style setting that is not behaving.

As to the conceptual question, perhaps you can help us reword the tooltip for the label "Why are some parameter names in bold?". It is currently: "When a setting's name is in bold it indicates that the value is different from what is in the stylesheet." to cover how you would want this explained to users.

Basically, the parameter is bolded if its value does not match what is is in the cascade of usfm_sb.sty, ptx2pdf.sty, custom.sty, ptxprint-mods.sty (ignoring those not selected from that list, etc.). In effect, it's bolded if ptxprint thinks you ever set the value to a new value in ptxprint rather than somewhere else.

**Comment by KimB2017** [comment]
> Having said that, we are toying with the idea of doing that for q/q1 and s/s1, but only because they are near synonyms and we would have to think about what happens if someone wants them different."

I think q/q1, etc., should be considered identical, and it should not need to be possible to format them differently. The reason for \q, \s and the like was to continue legacy practice (from pre-USFM days) and to give simplicity for those who do not need multiple levels. A project that intends to publish from USFM is "breaking the rules" if it considers \q and \q1 different. (mho)

**Comment by davidc86** [comment]
Thanks Kim for bringing that up. In fact, Publication Departments consider the USFM malformed if a publication contains both \s and \s1 in the same project. They also note that if there is ever a \s2 anywhere in the project, then all \s fields must become \s1 fields to pass their checks. 

So yes, actually \mt and \s and \is etc. are all legacy field markers. So \mt and \mt1 **must not be formatted differently**. Same with \is and \is1, and any other such level-based markers. 

It is preferred by GPS (or whatever its letters are these days) that \mt, \s, \is, etc. be retired. But what with so many USFM documents out there that use them, it would be a major headache to revise them all just to make this happen.

**Comment by davidc86** [comment]
Quick look and so far it appears the bolding of Style labels is acting as described.
This can probably be closed, though the issue of plain USFMs and USFMs with number 1, seems it might need thinking about.
